src/InstrumentationEngine.Lib/LoggerService.cpp

```diff
@@ -362,3 +415,53 @@ HRESULT CLoggerService::CreateSinks(vector<shared_ptr<ILoggerSink>>& sinks)
 
     return S_OK;
 }
+
+HRESULT CLoggerService::UpdateFlags(_In_ GUID classId, _In_ LoggingFlags loggingFlags)
+{
+    HRESULT hr = S_OK;
+
+    IfFailRetNoLog(UpdateFlagsInternal(
+        classId,
+        loggingFlags,
+        LoggingFlags_Errors));
+
+    IfFailRetNoLog(UpdateFlagsInternal(
+        classId,
+        loggingFlags,
+        LoggingFlags_Trace));
+
+    IfFailRetNoLog(UpdateFlagsInternal(
+        classId,
+        loggingFlags,
+        LoggingFlags_InstrumentationResults));
+
+    return S_OK;
+}
+
+HRESULT CLoggerService::UpdateFlagsInternal(_In_ GUID classId, _In_ LoggingFlags loggingFlags, _In_ LoggingFlags loggingLevel)
```

Comment:
>UpdateFlagsInternal [](start = 24, length = 19)

Adding a comment, for the specific LogLevel, we check if the InstrumentationMethod ClassId exists and whether it should Exist (ie. the loggingFlags includes the loggingLevel). If it exists and should not exist, we remove the classId from the set. If it doesn't exist and should exist, we add to the set.